hydra-cluster/test/Test/DirectChainSpec.hs

```diff
@@ -433,16 +445,20 @@ withDirectChainTest ::
 withDirectChainTest tracer config ctx action = do
   eventMVar <- newEmptyTMVarIO
 
-  let callback = \event -> do
-        atomically $ putTMVar eventMVar event
+  let callback event = atomically $ putTMVar eventMVar event
 
   wallet <- mkTinyWallet tracer config
 
-  withDirectChain tracer config ctx wallet initialChainState callback $ \Chain{postTx} -> do
+  withDirectChain tracer config ctx wallet initialChainState callback $ \Chain{postTx, draftCommitTx} -> do
     action
       DirectChainTest
         { postTx
         , waitCallback = atomically $ takeTMVar eventMVar
+        , draftCommitTx = \utxo -> do
+            eTx <- draftCommitTx utxo
+            case eTx of
+              Left e -> throwIO e
+              Right tx -> pure tx
```

Comment:
I had to do it because one of the tests use `shouldThrow` to assert proper exception is thrown now that we actually  use `draftCommitUTxO` in these tests.